ParatrooperUnity/Assets/Scripts/Damage.cs

```diff
@@ -11,7 +11,21 @@ private void OnTriggerEnter2D(Collider2D other)
 		if (health != null)
 		{
 			health.Damage(HP);
-			Destroy(gameObject);
+
+			DestroySelf();
+		}
+	}
+
+	private void DestroySelf()
+	{
+		var pooled = gameObject.GetComponent<Pooled>();
```

Comment:
use `RequiresComponent(typeof(Pooled))` attribute at the top and fetch the component in on Awake. It's a better form of dependency management, especially when behaviours need to be setup.